4_1_3 Securing the Platform (#169)
International-Data-Spaces-Association/IDS-RAM_4_0: documentation/4_Perspectives_of_the_Reference_Architecture_Model/4_1_Security_Perspective/4_1_3_Securing_the_Platform.md

# Securing the Platform

> Security of data processing in an IDS component depends on the system security of the utilized component. The hardware and software components that are critical to ensure the confidentiality and integrity of the transmitted and processed data form the Trusted Computing Base (TCB). Different deployment scenarios and their respective TCBs are described in the [first subsection](#deployment-scenarios). The [following subsection](#hardware-security-features) provides some background on hardware security features that might be used for securing the platform. Finally, the [last subsection](#platform-security-requirements) provides an overview of essential security requirements and how to achieve them.

Reviewer: Maybe in an IDS building block to avoid the duplicate term component?

Author: I fear that "IDS building block" is also a term utilized with a lot of different meanings in the IDS context. Instead, I've been trying to differentiate between "IDS Components" (connectors, Metadata Brokers, ...) and "hardware and software components" as the different parts utilized for building the software stack...
```suggestion
Security of data processing in an IDS Component depends on the system security of the utilized software stack consisting of different hardware and software components. Those hardware and software components that are critical for ensuring the confidentiality and integrity of the transmitted and processed data form the Trusted Computing Base (TCB). Different deployment scenarios and their respective TCBs are described in the [first subsection](#deployment-scenarios). The [following subsection](#hardware-security-features) provides some background on hardware security features that might be used for securing the platform. Finally, the [last subsection](#platform-security-requirements) provides an overview of essential security requirements and how to achieve them.
```

## Deployment Scenarios
In general, there are three different possibilities for the combinations of platform and connector service instances:
* A 1:1 mapping with a connector being a single device consisting of one platform instance and one connector service instance
* A 1:n mapping with multiple connectors on one physical device, i.e., a system with one platform instance and multiple connector service instances
* A n:m mapping with multiple devices offering a distributed deployment for multiple connector service instances (e.g. with Kubernetes)

The figure below shows the three options with their respective trusted computing bases:
![Deployment Scenarios](./media/deployment_scenarios.png)
#### _Fig. 4.1.3.1: Deployment Scenarios_

The TCB consists of the following components:
* All **hardware** components with access to unencrypted app data.
* **Firmware / bootloader / UEFI** representing all software components used to bootstrap the system and initialize the hardware before starting the kernel.

> * Optionally a **hypervisor** which may be used to better isolate multiple connectors on a device by providing a Virtual Machine (VM) for each of them. The impact on the TCB is illustrated for the 1:n mapping in a comparison to a solution with OS-level virtualization, i.e., containers.

Reviewer: Don't know how much we can trust a hypervisor in regards to isolation more than a container runtime.

Author: ```suggestion
* Optionally a **hypervisor** which may be used to isolate multiple connectors on a device by providing a Virtual Machine (VM) for each of them. The impact on the TCB is illustrated for the 1:n mapping in a comparison to a solution with OS-level virtualization, i.e., containers.
```

* The **kernel** connects user space software to the hardware of the device.
* A (container) **runtime** responsible for starting the execution of applications on the system.
* The **connector core services** taking care of essential connector functionalities as explained in the [System Layer](../../3_Layers_of_the_Reference_Architecture_Model/3_5_System_Layer/3_5_2_IDS_Connector.md).
* An arbitrary number of **apps** as introduced in the [System Layer](../../3_Layers_of_the_Reference_Architecture_Model/3_5_System_Layer/3_5_3_App_Store_and_Data_Apps.md).

Additionally, the TCB may include **external components** in the surrounding infrastructure of a connector which are used for security-relevant tasks, e.g. an authorization server.

> It is important to note that from a component security perspective there is no specific category for the usage of cloud solutions in comparison to an on-premise deployment. In both cases, the TCB includes all components that are required to ensure the confidentiality and integrity of the transmitted and processed data and needs to fulfill all requirements defined in the certification criteria catalog. The only differences lie in the responsibility for the different layers of the software stack and ensuring their conformity.

Reviewer: category is unclear to me in this context.

Author: ```suggestion
It is important to note that from a component security perspective there is no need to address the usage of cloud solutions differently than an on-premise deployment. In both cases, the TCB includes all components that are required to ensure the confidentiality and integrity of the transmitted and processed data and needs to fulfill all requirements defined in the certification criteria catalog. The only differences lie in the responsibility for the different layers of the software stack and ensuring their conformity.
```

## Hardware Security Features

In the following, we provide a brief overview on three hardware security features that might help fulfilling the requirements introduced in the [subsection below](#platform-security-requirements). For complete fulfillment of certification requirements, a combination of the mentioned techniques might need to be applied.

### Hardware Security Module (HSM)

> A Hardware Security Module (HSM) is a physical device containing one or more secure cryptoprocessor. It typically provides secure key generation, storage and management and supports using those keys for performing encryption or digital signing. Additionally, most HSMs provide tamper protection, ranging from a notification of detected tampering attempts to deleting the keys upon tamper detection. A HSM can either be directly integrated or attached to the hardware of one device or deployed as a network device which is shared by multiple clients in the network. The components necessary to manage and access the HSM belong to the TCB of the device using the HSM, even if they are deployed on other devices in the network (e.g. in case of a HSM as a network device).

Reviewer: ```suggestion
A Hardware Security Module (HSM) is a physical device containing one or more secure cryptoprocessor(s). It typically provides secure key generation, storage and management and supports using those keys for performing encryption or digital signing. Additionally, most HSMs provide tamper protection, ranging from a notification of detected tampering attempts to deleting the keys upon tamper detection. A HSM can either be directly integrated or attached to the hardware of one device or deployed as a network device which is shared by multiple clients in the network. The components necessary to manage and access the HSM belong to the TCB of the device using the HSM, even if they are deployed on other devices in the network (e.g. in case of a HSM as a network device).
```

* AMD offers a VM-based solution called Secure Encrypted Virtualization ([SEV](https://www.amd.com/en/processors/amd-secure-encrypted-virtualization)) with its newest version being [SEV-SNP](https://www.amd.com/system/files/TechDocs/56860.pdf) (Secure Nested Paging).
* After starting with the process-based Software Guard Extensions ([SGX](https://software.intel.com/content/www/us/en/develop/topics/software-guard-extensions.html)), Intel is now developing a VM-based solution called Trust Domain Extensions ([TDX](https://software.intel.com/content/www/us/en/develop/articles/intel-trust-domain-extensions.html)).
* Likewise, Arm is working on their own VM-based Confidential Computing Architecture ([CCA](https://www.arm.com/why-arm/architecture/security-features/arm-confidential-compute-architecture)).

To prove the correct setup of a TEE, each of the mentioned approaches offers some kind of remote attestation with proof of the integrity of the initial software components required. However, currently available implementations do not offer an attestation for the entire software stack, i.e., the TCB.

## Platform Security Requirements

> The Certification Criteria Catalog for Components defines various requirements for interoperability and security of an IDS connector. It provides three different Trust Levels with an increasing number of requirements as explained in [Section 4.2.4](../4_2_Certification_Perspective/4_2_4_Component_Certification.md).

Reviewer: ```suggestion
The Certification Criteria Catalog for Components defines various requirements for interoperability and security of an IDS Connector. It provides three different trust levels with an increasing number of requirements as explained in [Section 4.2.4](../4_2_Certification_Perspective/4_2_4_Component_Certification.md).
```

* **Isolation of processes**: To reduce the attack surface, interactions between different applications on the system shall be kept to a minimum and each application shall only get the privileges it requires. This can be achieved by isolating and restricting applications (containers) using security mechanisms in the Linux kernel such as namespaces, cgroups, AppArmor, SELinux, Linux capabilities and seccomp.
* **Tracking of events**: In case of security incidents, it is important to be able to reconstruct what happened on a connector. This is addressed by an integrity-protected audit log for all security-relevant events on the device. Logging should be performed on all layers: Within the platform up to the container runtime, e.g. by using a logging framework for Linux such as journald, and on the application layer. The latter can be done either by using a defined interface to the platform logging or by using an application-specific solution.

> * **Protecting utilized keys**: The cryptographic material (typically keys) used to achieve many of the requirements described above needs to be stored and used in a way that protects its confidentiality (for symmetric or private keys) or integrity (for public keys). Public keys used as root of trust for integrity and authenticity verification are typically provisioned onto the device during a secure initialization phase which is conducted in a secure environment. Private keys used by the device should be generated directly on the device and never directly displayed/provided to users or processes using them. Depending on the chosen Trust Level, the usage of hardware protection mechanisms such as, e.g., a HSM or TPM, is necessary.

Reviewer: Is "Trust Level" a proper noun that needs to be capitalised?

Author: from my perspective yes, since we introduce this terminology in the certification context and always use it to refer to those Trust Levels in IDS Certification.
